### ISSUE TITLE
`timeStamped` results don't make sense with `event.waitKeys` in pygame

### ISSUE THREAD

**Issue by stefanuddenberg** [psychopy/psychopy#2602]
Having `timeStamped` issues on a fresh install of Psychopy 3.2.0 on Windows 10 Pro build 18362. 
```python
key_list = ["left", "right"]
rt_clock = core.clock()
rt_clock.reset()
keys = event.waitKeys(keyList=key_list, timeStamped=rt_clock)
print(keys)
```

This ends up printing negative timestamps like: `keys = [['right', -7.34115810000003]]`
Using `timeStamped=True` does not give any timestamp at all, only showing the key pressed. Adding `clearEvents=True` as a parameter does not make any difference. This is all using the `pygame` for the `Window` object, as using `pyglet` causes an immediate crash (tested using `pyglet` version `1.4.3` and `1.3.2`).

**Comment by stefanuddenberg** [comment]
I may have fixed it by going into `event.py` and modifying `getKeys` at line 404 (changing the last argument to `psychopy.core.getTime()` instead of `0`:

```python
if havePygame and display.get_init():
    # see if pygame has anything instead (if it exists)
    for evts in evt.get(locals.KEYDOWN):
        # pygame has no keytimes --  but we can certainly still get them via psychopy!
        keys.append((pygame.key.name(evts.key), psychopy.core.getTime()))
```

**Comment by peircej** [comment]
Thanks for looking into this. A few points:

- Pygame isn't capable of doing any timestamping (it processes events fundamentally differently) so we never implemented that. I think it would be better if we simply provided a warning that `timestamped` should not be set to True in a pygame environment rather than provide a key that appears to have been asynchronously timestamped when it wasn't.
- the new Keyboard class (based on psychtoolbox engine, and requiring 64bit Python3+) is very much superior in its timing measurements to either pygame or pyglet. Take a look at the coder demo on  using that
- I'm not sure why you've got a crash using pyglet - for us that's still the preferred window system. Pygame is not officially supported and not something we test any more (maybe we should raise a warning about that too!). Could you telel me what error you get when pyglet crashes?

**Comment by stefanuddenberg** [comment]
Thanks, will look into the Keyboard class! As for the crash, I get no particular error message; the window pops up (blankly) and simply fails to respond.

**Comment by stefanuddenberg** [comment]
If it helps, I get the following output when quitting the program with Ctrl-C:

```
forrtl: error (200): program aborting due to control-C event
Image              PC                Routine            Line        Source

libifcoremd.dll    00007FFD3A113B58  Unknown               Unknown  Unknown
KERNELBASE.dll     00007FFD9FD85E73  Unknown               Unknown  Unknown
KERNEL32.DLL       00007FFDA1F57BD4  Unknown               Unknown  Unknown
ntdll.dll          00007FFDA228CE71  Unknown               Unknown  Unknown
```

**Comment by stefanuddenberg** [comment]
I've managed to fix(?) this (and TextStim centering issues) by using `pyglet==1.3.2` and the following window options (I am using a triple monitor setup):

```python
win = visual.Window(
    units="pix",
    fullscr=False,
    size=resolution,
    colorSpace="rgb",
    color=background_color,
    monitor="testMonitor",
    winType="pyglet",
    gammaErrorPolicy="ignore",
    allowGUI=False,
    screen=1,
)
```

**Comment by dvbridges** [comment]
Fixed in #2603
